src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,96 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private int size;
+    private T[] elements;
+
+    public ArrayList(int capacity) {
+        this.elements = (T[])new Object[capacity];
+    }
+
+    public ArrayList() {
+        this.elements = (T[])new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        increaseArraySizeIfFilled();
+        elements[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        ifIndexNotOutOfBonds(index, size());
+        increaseArraySizeIfFilled();
+        System.arraycopy(elements, index, elements, index + 1, size - index);
+        elements[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        ifIndexNotOutOfBonds(index, size() - 1);
+        return elements[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        ifIndexNotOutOfBonds(index, size() - 1);
+        elements[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        ifIndexNotOutOfBonds(index, size() - 1);
+        increaseArraySizeIfFilled();
```

Comment:
The call to 'increaseArraySizeIfFilled()' is unnecessary here. This method is meant to increase the size of the array when it's full, but during removal, the array size should not be increased.